[docs] Added docs for gossip encryption config (#5560)
hashicorp/consul: website/source/docs/platform/k8s/helm.html.md

  * <a name="v-global-datacenter" href="#v-global-datacenter">`datacenter`</a> (`string: "dc1"`) - The name of the datacenter that the agent cluster should register as. This may not be changed once the cluster is bootstrapped and running, since Consul doesn't yet support an automatic way to change this value.

>   * <a name="v-global-gossip" href="#v-global-gossip">`gossipEncryption`</a> - Provides the ability to enable [Gossip encryption](https://www.consul.io/docs/agent/encryption.html#gossip-encryption). Use of gossip encryption is highly recommended in production environments. You must specify both `secretName` and `secretKey` to enable gossip encryption.

Reviewer: ```suggestion
  * <a name="v-global-gossip" href="#v-global-gossip">`gossipEncryption`</a> - Provides the ability to enable [Gossip encryption](https://www.consul.io/docs/agent/encryption.html#gossip-encryption). We highly reccomend that you enable gossip encryption in all production environments. You must specify both `secretName` and `secretKey` to enable gossip encryption.
```

>   * <a name="v-global-gossip" href="#v-global-gossip">`gossipEncryption`</a> - Provides the ability to enable [Gossip encryption](https://www.consul.io/docs/agent/encryption.html#gossip-encryption). We highly reccomend that you enable gossip encryption in all production environments. You must specify both `secretName` and `secretKey` to enable gossip encryption.

Reviewer 2: ```suggestion
  * <a name="v-global-gossip" href="#v-global-gossip">`gossipEncryption`</a> - Provides the ability to enable [Gossip encryption](/docs/agent/encryption.html#gossip-encryption). We highly recommend that you enable gossip encryption in all production environments. You must specify both `secretName` and `secretKey` to enable gossip encryption.
```

  * <a name="v-global-datacenter" href="#v-global-datacenter">`datacenter`</a> (`string: "dc1"`) - The name of the datacenter that the agent cluster should register as. This may not be changed once the cluster is bootstrapped and running, since Consul doesn't yet support an automatic way to change this value.

>   * <a name="v-global-gossip" href="#v-global-gossip">`gossipEncryption`</a> - Configuration to enable [Gossip encryption](https://www.consul.io/docs/agent/encryption.html#gossip-encryption). Use of gossip encryption is highly recommended in production environments.

Reviewer 2: ```suggestion
  * <a name="v-global-gossip" href="#v-global-gossip">`gossipEncryption`</a> - Provides the ability to enable [Gossip encryption](https://www.consul.io/docs/agent/encryption.html#gossip-encryption). Use of gossip encryption is highly recommended in production environments.
```

Author: Fixed by 4e4fa3209219637ff88c22c65d8d9228546257e2.

>       - <a name="v-global-gossip-enabled" href="#v-global-gossip-enabled">`enabled`</a> (`boolean: true`) - If true, gossip traffic between agents will be encrypted.

Author: The `enabled` option has been removed entirely.

>       - <a name="v-global-gossip-secretname" href="#v-global-gossip-secretname">`secretName`</a> (`string: null`) - The name of the Kubernetes secret that contains the Gossip gncryption key. This defaults to null.

Author: Fixed by 4e4fa3209219637ff88c22c65d8d9228546257e2.
